src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java

```diff
@@ -263,14 +260,24 @@ public void writeBody(BufWriter b) {
                         int pos = b.size();
                         int lvSize = localVariables.size();
                         b.writeU2(lvSize);
+                        var lc = ((BufWriterImpl) b).labelContext();
                         for (LocalVariable l : localVariables) {
-                            if (!l.writeTo(b)) {
+                            int startBci = lc.labelToBci(l.startScope());
+                            int endBci = lc.labelToBci(l.endScope());
+                            if (startBci == -1 || endBci == -1) {
                                 if (context.deadLabelsOption() == Classfile.DeadLabelsOption.DROP_DEAD_LABELS) {
                                     lvSize--;
+                                    continue;
                                 } else {
                                     throw new IllegalArgumentException("Unbound label in local variable type");
                                 }
                             }
+                            int length = endBci - startBci;
+                            b.writeU2(startBci);
+                            b.writeU2(length);
+                            b.writeIndex(l.name());
+                            b.writeIndex(l.type());
+                            b.writeU2(l.slot());
```

Comment:
This idiom is repeated four times; seems like it can be factored somewhere into a helper?